File: src/main/java/com/hedvig/backoffice/services/account/dto/SchedulerStateDto.java
package com.hedvig.backoffice.services.account.dto;

import com.hedvig.backoffice.services.account.ChargeStatus;

import lombok.Value;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.util.Optional;
import java.util.UUID;

@Value
public class SchedulerStateDto {
  UUID stateId;
  String memberId;
  ChargeStatus status;
  String changedBy;
  Instant changedAt;
  Optional<MonetaryAmount> amount;
  Optional<UUID> transactionId;
}

File: src/main/java/com/hedvig/backoffice/services/account/dto/SchedulerStateDto.kt
package com.hedvig.backoffice.services.account.dto

import com.hedvig.backoffice.services.account.ChargeStatus
import java.time.Instant
import java.util.UUID
import javax.money.MonetaryAmount

data class SchedulerStateDto(
    var stateId: UUID,
    var memberId: String,
    var status: ChargeStatus,
    var changedBy: String,
    var changedAt: Instant,
    var amount: MonetaryAmount?,
    var transactionId: UUID?
)
